include/deal.II/base/work_stream.h

```diff
@@ -1227,6 +1291,109 @@ namespace WorkStream
         chunk_size);
   }
 
+
+  template <typename MainClass,
+            typename Iterator,
+            typename ScratchData,
+            typename CopyData>
+  void
+  run(const typename IteratorRange<
+        typename identity<Iterator>::type>::IteratorOverIterators &begin,
```

Comment:
I think you probably want to let the first occurrence of `Iterator` here be without the `identity` thing. The compiler can work things out from the uses of the type below, but it might be more obvious if you used the type directly here.

Reply:
I got stuck with this because the template-related error I was getting when moving on to finally adding support for `mesh_loop()` were horrendous. However, your tips here and below are really useful, so maybe I'll finally be able to come back to this and find a solution to the issue I was having.